evalml/data_checks/ts_splitting_data_check.py

```diff
@@ -0,0 +1,93 @@
+    on all potential outcomes which will cause errors during prediction.
+
+    Args:
+        problem_type (str or ProblemTypes): Problem type.
+        n_splits (int): Number of time series splits.
+    """
+
+    def __init__(self, problem_type, n_splits):
+        self.problem_type = problem_type
+        if handle_problem_types(self.problem_type) not in [
+            ProblemTypes.TIME_SERIES_BINARY,
+            ProblemTypes.TIME_SERIES_MULTICLASS,
+        ]:
+            raise ValueError(
+                "Valid splitting of labels in time series is only defined for time series binary and time series multiclass problem types."
+            )
+        self.n_splits = n_splits
+        self._splitter = SkTimeSeriesSplit(n_splits=self.n_splits)
+
+    def validate(self, X, y):
+        """Check if the training and validation targets are compatible with time series data splitting.
+
+        Args:
+            X (pd.DataFrame, np.ndarray): Ignored. Features.
+            y (pd.Series, np.ndarray): Target data.
+
+        Returns:
+            dict: dict with a DataCheckError if splitting would result in inadequate class representation.
+
+        Example:
+            >>> import pandas as pd
+            ...
+            >>> X = None
+            >>> y = pd.Series([0 if i < 45 else i % 2 if i < 55 else 1 for i in range(100)])
+            ...
+            >>> ts_splitting_check = TimeSeriesSplittingDataCheck("time series binary", 3)
+            >>> assert ts_splitting_check.validate(X, y) == {
+            ...     "errors": [{'message': 'Time Series Binary and Time Series Multiclass problem '
+            ...                             'types require every training and validation split to '
+            ...                             'have at least one instance of all the target classes. '
+            ...                             'The following splits are invalid: [1, 3]',
+            ...                  'data_check_name': 'TimeSeriesSplittingDataCheck',
+            ...                  'level': 'error',
+            ...                  'details': {'columns': None, 'rows': None, 'invalid_splits': {1: {"Training": [0, 25]},
+            ...                                                                                3: {"Validation": [75, 100]}}},
+            ...                  'code': 'TIMESERIES_TARGET_NOT_COMPATIBLE_WITH_SPLIT'}],
+            ...     "warnings": [],
+            ...     "actions": []}
+        """
+        results = {"warnings": [], "errors": [], "actions": []}
+
+        y = infer_feature_types(y)
+
+        invalid_splits = {}
+        if y is not None:
+            for split_num, (train, val) in enumerate(self._splitter.split(X=y)):
+                invalid_dict = {}
+                train_targets = y[train]
+                val_targets = y[val]
+                if train_targets.nunique() < y.nunique():
```

Comment:
I think we can call this once? I believe pandas doesn't cache the result so we're redoing all the work every time.

Reply:
good call!